src/coreclr/inc/gcinfotypes.h

```diff
@@ -53,7 +53,7 @@ inline UINT32 CeilOfLog2(size_t x)
     return result;
 }
 
-enum GcSlotFlags
+enum GcSlotFlags : unsigned short
```

Comment:
Is this fixing any warnings, or is it just an unrelated change?

Reply:
This is fixing warnings on usage. One of the hashing mechanisms is assumes the flags will fit into an `unsigned short` and is types as such. By defining it here that assumption flows nicely.

Reply:
Yes, but it is creating inconsistencies in the enum definitions. We tend to prefer to use the default enum underlying type. It makes this enum to look different from the other enums next to it. And there is mirror of this enum in C# that would need to be updated too. 

If it is just one place in hashing, I would add a cast there instead.

Reply:
I would much rather properly define the type to be honest. The expressiveness exists so I'd argue we should use it. I see how many things would need to change.

Reply:
There would be 7 seemingly random casts in the `gcencode.cpp` file. I really dislike this design here. We are adding far more confusion than clarity when we should be expressing our intent clearly with the type-system that exists.

Reply:
I think I have a compromise.

Reply:
FWIW, https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-underlying says "Specify the underlying type of an enumeration only when necessary". It is not obvious to me that avoiding one cast in hashing qualifies as necessary.

Reply:
Agreed for `enum class`. I don't see the same argument for `enum` types that grow and transparently cast to the underlying type. The `enum class` doesn't permit this so casting is required and thus frowned upon.